### PR TITLE
Fix 'imageEdgeInsets' was deprecated in iOS 15.0

### PR DESCRIPTION
**Pull request by iamgabrielma**
Based on https://github.com/woocommerce/woocommerce-ios/pull/8094 , that PR should be merged first.

### Description
By updating the minimum release version to iOS 15 recently, we've seen some deprecation warnings. This PR fixes the case for `'imageEdgeInsets' was deprecated in iOS 15.0: This property is ignored when using UIButtonConfiguration`

On some buttons, we rely on the `distributeTitleAndImage(_:)` extension in order to add the correct padding between image and text in a UIButton. However, since we're moving to use `UIButton.Configuration` now, this functionality comes for free by calling `.imagePadding() ` instead. 

### Changes
- Deleted `UIButton+TitleAndImage`
- Added `.imagePadding()` usage to the 4 cases where we use this extension.

With the change, we can also change how we do other things, like setting title or image, but this is outside of the scope of this PR, which only intends to remove the Xcode deprecation warnings. We'll leave that for a different PR :) 

### Testing instructions
Confirm that the following buttons still keep the same separation between Image and Text:
- **Case 1:** Go to Orders > open an Order > Check the "+ Add Customer Note" and "+ "Add Note" buttons.
- **Case 2:** Go to Orders > open a "pending payment" Order > Tap on "Collect Payment" > Select "Card" > See the "Learn More about In-Person Payments" 
- **Case 3:** Go to Products > Tap on a Product > See the "+ Add more details" at the bottom of the view.

| Case 1 | Case 2 | Case 3 |
|-|-|-|
|  |  |  |

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/CardPresentPayments/CardPresentPaymentsModalViewController.swift [hunk 1]
```diff
@@ -281,7 +281,9 @@ private extension CardPresentPaymentsModalViewController {
     func configureAuxiliaryButton() {
 
         guard shouldShowAuxiliaryButton() else {
-            auxiliaryButton.isHidden = true
+            // Only for testing, switch back to true before merging
+            //auxiliaryButton.isHidden = true
+            auxiliaryButton.isHidden = false
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/CardPresentPayments/CardPresentPaymentsModalViewController.swift]: This is for convenience when testing the Case 2, has to be removed before merging.

File: WooCommerce/Classes/ViewRelated/CardPresentPayments/CardPresentPaymentsModalViewController.swift [hunk 2]
```diff
@@ -293,7 +295,9 @@ private extension CardPresentPaymentsModalViewController {
             auxiliaryButton.setAttributedTitle(viewModel.auxiliaryAttributedButtonTitle, for: .normal)
             auxiliaryButton.setImage(viewModel.auxiliaryButtonimage, for: .normal)
             if viewModel.auxiliaryButtonimage != nil {
-                auxiliaryButton.distributeTitleAndImage(spacing: 8.0)
+                var config = UIButton.Configuration.plain()
+                config.imagePadding = CGFloat(Constants.buttonTitleAndImageSpacing)
```
Comment (ealeksandrov) [inline WooCommerce/Classes/ViewRelated/CardPresentPayments/CardPresentPaymentsModalViewController.swift]: nit: `buttonTitleAndImageSpacing` is already declared as `CGFloat`, no need to wrap/convert.
Applies also to 3 similar cases below.

**Review by ealeksandrov — COMMENTED** [review]
Code looks good and described test cases work well!
But I've found one regression in `BottomButtonContainerView`:


There is a conflict between 2 `UIButton.Configuration`s, so "primary button" loses its custom background and white text blends with white superview.

File: WooCommerce/Classes/ViewRelated/CardPresentPayments/CardPresentPaymentsModalViewController.swift (hunk 2 above)
Comment (iamgabrielma) [reply WooCommerce/Classes/ViewRelated/CardPresentPayments/CardPresentPaymentsModalViewController.swift]: > nit: `buttonTitleAndImageSpacing` is already declared as `CGFloat`, no need to wrap/convert. Applies also to 3 similar cases below.

Oops! Removed on 3d9760bd7208e955a12daa8b56673924a97b530c

**Comment by iamgabrielma** [comment]
> There is a conflict between 2 `UIButton.Configuration`s, so "primary button" loses its custom background and white text blends with white superview.

Good catch! I didn't see it as I was testing it in Dark Mode 🙇 

This has been changed on e67f70ec0f7985e2630f0fd3b0a1d85fef355f78 by leaving the `UIButton.Configuration` responsibility to `applyPrimaryButtonStyle()`, then overriding only the cases when `applyLinkButtonStyle()` has an image. As this instance is only used by the `ProductFormViewController`, for now I preferred to leave it there rather than potentially modifying the button configuration to all buttons that apply the link style.

**Review by ealeksandrov — APPROVED** [review]
LGTM! :shipit:
One note: can we merge it as a betafix to `release/11.4`, so we fix this minor alignment issue before reaching the users?

**Pull request merged**